database-commons/src/main/java/co/cask/db/batch/sink/AbstractDBSink.java

```diff
@@ -102,22 +102,18 @@ public void prepareRun(BatchSinkContext context) {
               dbSinkConfig.jdbcPluginName,
               connectionString, "columns");
 
+    Schema inputSchema = context.getInputSchema();
+    Objects.requireNonNull(inputSchema, "AbstractDBSink requires a single known schema.");
```

Comment:
Now that we are looking up the schema, we should allow the input schema to be null. If it is null, try to use whatever schema the output table has.

Reply:
I'm not sure, how can Sink be used without explicit schema. I mean, it should always be connected to either source or transform, which have output schema that is passed to sink. Or am I getting something wrong?

Reply:
A null schema happens if the previous stage set their output schema to be null. It indicates that the schema is either not known or that it can be multiple schemas. This is normally not the case, but we have seen instances where the previous stage might sometimes emit records with fields a,b,c and might sometimes emit records with fields a,b,d. In that scenario, it's possible they might want to connect it to a db sink where the table has columns a,b,c,d.

It's not a common scenario, but it looks like we could support that without much work.